File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function isPrime(num) {
    if (num < 2) 
        return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) 
            return false;
    }
    return true;
}

function permutate(arr, str = '', permutations) {
    if (str.length > 0) 
        permutations.add(parseInt(str));
  
    for (let i = 0; i < arr.length; i++) {
        let copy = [...arr];
        copy.splice(i, 1);
        permutate(copy, str + arr[i], permutations);
    }
}

function solution(numbers) {
    const numArray = numbers.split('');
    let permutations = new Set();
  
    permutate(numArray, '', permutations);

    let count = 0;
    permutations.forEach(num => {
        if (isPrime(num)) count++;
    });
  
    return count;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <set>
using namespace std;

bool isPrime(int n) 
{
    if (n < 2) 
        return false;
    
    for (int i = 2; i * i <= n; i++)
        if (n % i == 0) 
            return false;
    
    return true;
}

void permutate(const string& numbers, set<int>& nums, string current = "", vector<bool> used = {}) 
{
    if (used.empty()) used.resize(numbers.length(), false);

    if (!current.empty())
        nums.insert(stoi(current));

    for (int i = 0; i < numbers.length(); i++) 
    {
        if (!used[i]) 
        {
            used[i] = true;
            permutate(numbers, nums, current + numbers[i], used);
            used[i] = false;
        }
    }
}

int solution(string numbers) 
{
    set<int> nums;
    
    permutate(numbers, nums);
    
    int count = 0;
    for (auto num : nums)
        if (isPrime(num)) count++;
    
    return count;
}